src/main/java/antlr/TreepatVisitorImplementationFunctions.kt

```diff
@@ -0,0 +1,109 @@
+package antlr
+
+import antlr.generate.TreepatParser
+import antlr.generate.TreepatVisitor
+import functions.ChildFunction
+import functions.NodeFunction
+import functions.SiblingFunction
+import org.antlr.v4.runtime.tree.ErrorNode
+import org.antlr.v4.runtime.tree.ParseTree
+import org.antlr.v4.runtime.tree.RuleNode
+import org.antlr.v4.runtime.tree.TerminalNode
+import tree.TargetTreeNode
+import java.util.stream.Collectors
+
+class TreepatVisitorImplementationFunctions : TreepatVisitor<(TargetTreeNode) -> List<TargetTreeNode>> {
+    override fun visitModel(ctx: TreepatParser.ModelContext?): (TargetTreeNode) -> List<TargetTreeNode> {
+        return ctx!!.subtree().accept<(TargetTreeNode) -> List<TargetTreeNode>>(this)
+    }
+
+    override fun visitSubtree(ctx: TreepatParser.SubtreeContext?): (TargetTreeNode) -> List<TargetTreeNode> {
+        val expression = ctx!!.expression().accept<(TargetTreeNode) -> List<TargetTreeNode>>(this)
+        if (ctx.child() == null) {
+            return expression
+        }
+        val child = ctx.child().accept<(TargetTreeNode) -> List<TargetTreeNode>>(this)
+        return ChildFunction().child(father = expression, child = child)
+    }
+
+    override fun visitExpression(ctx: TreepatParser.ExpressionContext?): (TargetTreeNode) -> List<TargetTreeNode> {
+        return if (ctx!!.simple_expression() != null) {
+            ctx.simple_expression().accept<(TargetTreeNode) -> List<TargetTreeNode>>(this)
+        } else ctx.depth_closure().accept<(TargetTreeNode) -> List<TargetTreeNode>>(this)
+    }
+
+    override fun visitChild(ctx: TreepatParser.ChildContext?): (TargetTreeNode) -> List<TargetTreeNode> {
+        return ctx!!.sibling().accept<(TargetTreeNode) -> List<TargetTreeNode>>(this)
+    }
+
+    override fun visitSibling(ctx: TreepatParser.SiblingContext?): (TargetTreeNode) -> List<TargetTreeNode> {
+        val siblings: List<(TargetTreeNode) -> List<TargetTreeNode>>
+        siblings = ctx!!.union().stream()
+            .map { instruction: TreepatParser.UnionContext -> instruction.accept<(TargetTreeNode) -> List<TargetTreeNode>>(this) }
+            .collect(Collectors.toList())
+        return SiblingFunction().sibling(siblings)
+    }
+
+    override fun visitUnion(ctx: TreepatParser.UnionContext?): (TargetTreeNode) -> List<TargetTreeNode> {
+        val nodes: List<(TargetTreeNode) -> List<TargetTreeNode>>
+        nodes = ctx!!.subtree_wrapper().stream()
+            .map { node: TreepatParser.Subtree_wrapperContext -> node.accept<(TargetTreeNode) -> List<TargetTreeNode>>(this) }
+            .collect(Collectors.toList())
+        return nodes[0]
+    }
+
+    override fun visitSubtree_wrapper(ctx: TreepatParser.Subtree_wrapperContext?): (TargetTreeNode) -> List<TargetTreeNode> {
```

Comment:
Do you think we could change the naming convention of the grammar in the file `Treepat.g4` from *snake* to *Camel Case* so that the methods will be in *Camel Case*. I know is stupid 😒 but it will help with consistency 😃.

![](https://media.giphy.com/media/Q94xQWspTUkShljj8P/giphy.gif)

Reply:
Jajajaj okay I will check the grammar and do the change